packages/gatsby-source-filesystem/src/create-remote-file-node.js

```diff
@@ -196,22 +198,31 @@ async function processRemoteNode({
 
   // Create the temp and permanent file names for the url.
   const digest = createHash(url)
-  const ext = getRemoteFileExtension(url)
+  if (!ext){
+    ext = getRemoteFileExtension(url)
+  }
 
   const tmpFilename = createFilePath(programDir, `tmp-${digest}`, ext)
-  const filename = createFilePath(programDir, digest, ext)
 
   // Fetch the file.
   try {
     const response = await requestRemoteNode(
       url,
       headers,
       tmpFilename,
-      filename
     )
     // Save the response headers for future requests.
-    cache.set(cacheId(url), response.headers)
-
+    await cache.set(cacheId(url), response.headers)
+
+    // If the user did not provide an extension and we couldn't get one from remote file, try and guess one
+    if (ext === ``) {
+      const buffer = readChunk.sync(tmpFilename, 0, 4100)
```

Comment:
Where's 4100 coming from? What's the significance?

If we have a "magic number" let's encapsulate it like so, e.g.

```js
const MAX_BUFFER_SIZE = 4100

// later

const buffer = readChunk.sync(tmpFilename, 0, MAX_BUFFER_SIZE)
```

Reply:
I think we should use constant provided by `file-type` lib (`fileType.minimumBytes`):
```
const readChunk = require('read-chunk');
const fileType = require('file-type');
 
const buffer = readChunk.sync('unicorn.png', 0, fileType.minimumBytes);
 
fileType(buffer);
//=> {ext: 'png', mime: 'image/png'}
```

Reply:
heheh it was added 4 days ago :D i admit i just c/p the part with the buffer size its true that i had to avoid the magic number.

i ll correct this and mention the changes in the docs.